planner: prepared plan cache support limit (#12307)
pingcap/docs: system-variables.md

- Persists to cluster: Yes
- Type: Boolean
- Default value: `ON`
- This variable specifies whether to use the pipeline execution algorithm for window functions.

### `tidb_enable_plan_cache_for_param_limit` <span class="version-mark">New in v6.6.0</span>

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Type: Boolean
- Default value: `ON`

> - This variable controls whether Prepared Plan Cache caches execution plans with `Count` being `?` after `Limit`. The default value is `ON`, which means Prepared Plan Cache supports caching such execution plans. Note that Prepared Plan Cache does not support caching execution plans with a `Count` that is greater than 10000.

Author: ```suggestion
- This variable controls whether Prepared Plan Cache caches execution plans with a variable as the `LIMIT` parameter (`LIMIT ?`). The default value is `ON`, which means Prepared Plan Cache supports caching such execution plans. Note that Prepared Plan Cache does not support caching execution plans with a variable that is greater than 10000.
```

Author: @qw4990 @Oreoxmt @fzzf678 PTAL

Reviewer: LGTM
